packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -333,7 +328,7 @@ export class AuthUtil {
             // Edge Case: With the addition of Amazon Q/Chat scopes we may need to add
             // the new scopes to existing pre-chat connections.
             if (addMissingScopes) {
```

Comment:
Since BiD and IdC are at parity, we only need to add Q scopes. We do not need the separate BiD vs IdC checks anymore.

```suggestion
            if (addMissingScopes && isSsoConnection(this.conn) && !isValidAmazonQConnection(this.conn)) {
                const conn = await this.secondaryAuth.addScopes(this.conn, amazonQScopes)
                await this.secondaryAuth.useNewConnection(conn)
                return
           }
```